examples/tutorials/advanced/Federated Dataset.ipynb

```diff
@@ -0,0 +1,460 @@
+    "        \n",
+    "        #<--If any transforms have to be performed on the dataset\n",
+    "        self.transform = transform\n",
+    "        \n",
+    "        \n",
+    "    def to_torchtensor(self):\n",
+    "        \n",
+    "        \"Transform Numpy Arrays to Torch tensors\"\n",
+    "        \n",
+    "        self.data=torch.from_numpy(self.data)\n",
+    "        self.labels=torch.from_numpy(self.targets)\n",
+    "    \n",
+    "        \n",
+    "    def __len__(self):\n",
+    "        \n",
+    "        \"\"\"Required Method\n",
+    "            \n",
+    "           Returns:\n",
+    "        \n",
+    "                Length [int]: Length of Dataset/batches\n",
+    "        \n",
+    "        \"\"\"\n",
+    "        \n",
+    "        return len(self.data)\n",
+    "    \n",
+    "\n",
+    "    def __getitem__(self, idx):\n",
+    "        \n",
+    "        \"\"\"Required Method\n",
+    "        \n",
+    "           The output of this method must be torch tensors since torch tensors overloaded \n",
+    "           with share() method which is used to share data to workers.\n",
+    "            \n",
+    "           Returns:\n",
+    "                 \n",
+    "                 Data [Torch Tensor]: \n",
+    "                 Target [ Torch Tensor]:\n",
+    "        \n",
+    "        \"\"\"\n",
+    "        \n",
+    "        sample=self.data[idx]\n",
+    "        target=self.targets[idx]\n",
+    "                \n",
+    "        if self.transform:\n",
+    "            sample = self.transform(sample)\n",
+    "\n",
+    "        return sample,target"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "federated_train_loader = sy.FederatedDataLoader( # <-- this is now a FederatedDataLoader \n",
+    "                         SVHNDataset(X_train,y_train)\n",
+    "                         .federate((bob, alice)),batch_size=args.batch_size)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<h2>Syft Base Datasets</h2>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Syft Base Dataset is a simplified dataset feature of Syft Library that allows you create datasets by simply providing training data and corresponding labels. This could also be utilized in federated data loaders. Ensure the inputs to BaseDataset are torch tensors. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "base=sy.BaseDataset(torch.from_numpy(X_train),torch.from_numpy(y_train))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "federated_train_loader = sy.FederatedDataLoader( # <-- this is now a FederatedDataLoader \n",
```

Comment:
Maybe distinguish the two phases: 1) .federate and 2) put into a federatedDataLoader